thetanuts/thetanuts/config.py

```diff
@@ -0,0 +1,277 @@
+        #     }
+        #    ), private_key
+        #  ).rawTransaction
+        # )
+
+        # Initiate new round details for ParadigmBridge
+        if (
+            bridgeContract.functions.vaultNextStrikeX1e6(oToken).call() == 0
+        ):  # New round not configured yet
+            amtToSell = vaultContract.functions.initNewRound([int(10000e6)], 0, 0).call(
+                {"from": vaultContract.functions.designatedMaker().call()}
+            )  # Get active balance in vault - will fail if not ready
+            price = 2000e6  # Strike Price to sell at (multiplied by 1 million)
+            tx = w3.eth.send_raw_transaction(
+                w3.eth.account.sign_transaction(
+                    bridgeContract.functions.setNextStrikeAndSize(
+                        oToken, int(price), amtToSell
+                    ).build_transaction(
+                        {'nonce': Nonce(nonce), 'from': public_key, "gas": 200000}
+                    ),
+                    private_key,
+                ).rawTransaction
+            )
+            print("Sent OWNER transaction for setting new strike and size", tx.hex())
+            w3.eth.wait_for_transaction_receipt(tx)
+            time.sleep(5)
+        return str(int(oToken[2:], 16))
+
+    def get_otoken_details(
+        self,
+        *,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        **kwargs: Any,
+    ) -> OfferTokenDetails:
+        """Return details about the offer token"""
+
+        w3 = web3.Web3(web3.HTTPProvider(rpc_uri))
+        bridgeContract = w3.eth.contract(
+            w3.toChecksumAddress(contract_address),
+            abi=json.load(open("thetanuts/abis/ParadigmBridge.json", "r")),
+        )
+
+        aucDetails = bridgeContract.functions.getAuctionDetails(kwargs["oToken"]).call()
+
+        return {
+            "collateralAsset": aucDetails[0],
+            "underlyingAsset": aucDetails[1],
+            "strikeAsset": aucDetails[2],
+            "strikePrice": aucDetails[3],
+            "expiryTimestamp": aucDetails[4],
+            "isPut": aucDetails[5],
+        }
+
+    def get_offer_details(
+        self,
+        *,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        offer_id: int,
+        **kwargs: Any,
+    ) -> OfferDetails:
+        """Return details for a given offer"""
+
+        w3 = web3.Web3(web3.HTTPProvider(rpc_uri))
+        bridgeContract = w3.eth.contract(
+            w3.toChecksumAddress(contract_address),
+            abi=json.load(open("thetanuts/abis/ParadigmBridge.json", "r")),
+        )
+        vault_address = w3.toChecksumAddress("0x%040x" % offer_id)
+
+        try:
+            aucDetails = bridgeContract.functions.getAuctionDetails(vault_address).call()
+        except Exception:
+            raise ValueError("The argument is not a valid offer")
+
+        return {
+            'seller': vault_address,
+            'oToken': vault_address,
```

Comment:
Is it correct that vault_address is returned for both seller and oToken?

Reply:
yup! the contract itself directly sells, we dont have a opyn intermediary token

Reply:
Got it, thank you for the confirmation